lsp-mode.el

```diff
@@ -5218,6 +5218,23 @@ MODE is the mode used in the parent frame."
         (lambda (_start _end _match) t))
   (prettify-symbols-mode 1))
 
+(defvar lsp--markdown-link-map
```

Comment:
can you make this public? I guess one might wanna change the bindings to C-RET, or Ctrl-Click.

Reply:
Done. Renamed into a more friendly `lsp-help-link-keymap`.

I just realized that there are several url types in markdown: `markdown-plain-url-face` and `markdown-url-face`. 

I am afraid we are getting too much into markdown's internals with this. This is necessary because the exposed  markdown API like  `markdown-follow-link-at-point` and `markdown-link-p` breaks in `*lsp-help*` buffer. 

It breaks because `lsp` removes `invisible` text in [lsp--buffer-string-visible](https://github.com/emacs-lsp/lsp-mode/blob/1cdb46c26a137be087e374cb2496bb909c7b181f/lsp-mode.el#L5246) which is relied upon by markdown internals. Is that  invisible-text-removal really necessary? If not, things could be simplified a bit.

Reply:
> It breaks because `lsp` removes `invisible` text in [lsp--buffer-string-visible](https://github.com/emacs-lsp/lsp-mode/blob/1cdb46c26a137be087e374cb2496bb909c7b181f/lsp-mode.el#L5246) which is relied upon by markdown internals. Is that invisible-text-removal really necessary?

It is needed for lsp-ui-sideline. The overlays does not handle invisible text.